fcli-ssc/src/main/java/com/fortify/cli/ssc/role/helper/SSCRoleHelper.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * (c) Copyright 2020 Micro Focus or one of its affiliates
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a 
+ * copy of this software and associated documentation files (the 
+ * "Software"), to deal in the Software without restriction, including without 
+ * limitation the rights to use, copy, modify, merge, publish, distribute, 
+ * sublicense, and/or sell copies of the Software, and to permit persons to 
+ * whom the Software is furnished to do so, subject to the following 
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included 
+ * in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY 
+ * KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE 
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR 
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS 
+ * IN THE SOFTWARE.
+ ******************************************************************************/
+package com.fortify.cli.ssc.role.helper;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fortify.cli.common.json.JsonHelper;
+import com.fortify.cli.common.rest.runner.UnexpectedHttpResponseException;
+import com.fortify.cli.ssc.rest.SSCUrls;
+import kong.unirest.GetRequest;
+import kong.unirest.UnirestInstance;
+
+import javax.validation.ValidationException;
+
+public class SSCRoleHelper {
+    public static final SSCRoleDescriptor getRole(UnirestInstance unirestInstance, String roleNameOrId, String... fields) {
+        GetRequest request = unirestInstance.get(SSCUrls.ROLE(roleNameOrId));
+
+        try{
```

Comment:
I think role id's are not numeric, right? So you can't tell beforehand whether the user passed a role name or id?
In that case, indeed you'd probably need to call the two endpoints like you're currently doing, but you may want to try combining these in a single request using `SSCBulkRequestBuilder`

Reply:
Done.

Reply:
**Done.**
But getting this to work wasn't as straightforward as I hoped it would be. The `SSCBulkRequestBuilder` was pretty easy to use, but there is an odd behavior that I observed when interacting with the SSC bulk API.

When using the bulk request to get information from the `/api/v1/roles` api, if the `q` query parameter is used, I observe that:
-  encoded double quote characters can't be used to surround a string
-  If you use the request builder's `queryParameter()` method, any spaces will be replaced with the `+` character, which causes issues. To avoid any issues, for values supplied to the `name` parameter, a space needs to be encoded as `%20`, not the `+` character. 

So as an example:
- Making a standard GET request to the following url will work, but if I take this exact same url and use it in a bulk request, no data will be retrieved:
  -  `http://192.168.65.213:8080/ssc/api/v1/roles?q=name%3A%22Security+Lead%22&limit=2`

- Instead, I need to ensure that the url is modified to look like the following in order to successfully return data from a bulk API call:
  -  `http://192.168.65.213:8080/ssc/api/v1/roles?q=name:Security%20Lead&limit=2`